packages/indexer-agent/src/agent.ts

```diff
@@ -877,7 +888,12 @@ class Agent {
       )
     }
 
-    const lifetime = Math.max(1, maxAllocationEpochs - 1)
+    const lifetime = Math.max(
+      1,
+      rule?.allocationMaxLifetime
+        ? rule?.allocationMaxLifetime - 1
+        : maxAllocationEpochs,
```

Comment:
`maxAllocationEpochs - 1`?

Reply:
So from the above [calculation for `allocationLifetime`](https://github.com/graphprotocol/indexer/pull/343/files#diff-186fe48e2f955474051fa281f7fd59e1a31551cbe1ea7f3ef4ec6640342c1f92L696) where `lifetime = max epoch - 1`, I was concerned if we want to use `rule.allocationMaxLifetime` for agent to close right before or at the maximum lifetime epoch?

Reply:
What do you think about changing the rule property to `allocationLifetime` and if it's not supplied have it default to `maxAllocationEpochs` - 1? Then the behavior would be more explicit and we can just use that value rather than checking if it exists etc...

Reply:
yes, that makes sense! will update it